### ISSUE TITLE
`local -n` and `declare -A` are invalid options in OS X

### ISSUE THREAD

**Issue by stevenhaddox** [robbycuenot/unifi-tf-generator#4]
I'm running OS X Sonoma and am receiving the errors in the subject from the `scripts/mapping_utils.sh` and `scripts/json_utils.sh` scripts. I've tried this in zsh, bash, and homebrew's bash. I love the idea of this project, but unfortunately am stuck with limited time and won't get a chance to play with it for awhile. I did want to share the errors I encountered though in case it's of any interest for others.

**Comment by robbycuenot** [comment]
Hey @stevenhaddox, thanks for testing. I just verified your issue on a 2018 Mac Mini on Sonoma. Turns out that Mac OS still ships with bash 3.2...

I've pushed a PR to address this: #5 

In short, you'll need to pull the latest changes from this repo, install bash with brew, launch the newer version of bash (run `/usr/local/Cellar/bash/*/bin/bash` in the terminal), and then the scripts should work as expected. Tested and verified on my machine. Let me know if you get a chance to try it!

**Comment by robbycuenot** [comment]
@stevenhaddox I should note, that the reason you were seeing issues even with the `brew` installed version of bash, is that I had a shebang at the start of each `.sh` file that pointed to `#!/bin/bash`. This meant that even if you launched the script from a newer bash version, subsequent calls to child scripts would be invoked with the 3.2 version in Mac OS's /bin folder. I've removed the shebang everywhere, as bash seems to know what to do without it and it was only causing problems.

**Comment by stevenhaddox** [comment]
Sorry for the delayed reply. This fix seems to have worked a treat. 

Also, I think you may be able to keep the she-bang as long as you utilize `#!/usr/bin/env bash` rather than `#!/usr/bin/bash`. I believe this basically tells the script to look at the $PATH for a bash prior to utilizing the specific bash that's hardcoded as you originally had it.

I've confirmed this works locally by modifying the `script/get_token.sh` script to have the first three lines and output:
```
#!/usr/bin/env bash
echo `which bash`
echo `bash --version`
```

```
# Output from the above:
#=> /opt/homebrew/bin/bash
#=> GNU bash, version 5.2.26(1)-release (aarch64-apple-darwin23.2.0) [... truncated ...]
#=> Logging in...
```

The main benefit I see of using a she-bang that points to the `/usr/bin/env` is that this allows the scripts to execute as expected in another terminal as well (e.g., I use zsh normally and it all worked as expected within zsh without having to start a new bash session manually).

**Comment by robbycuenot** [comment]
@stevenhaddox I was debating that approach, though I wasn't sure if it would cause similar issues (for example, someone having bash 3 as a higher priority in the path over bash 5). This scenario is pretty unlikely though. This fix is almost exclusively for Mac OS, so if it works there with the brew version of bash I'd say it makes sense. I'm going to test and merge. Thanks for your contribution!

**Comment by robbycuenot** [comment]
Tested and merged PR #6
